### ISSUE TITLE
Messages not getting through to both sides

### ISSUE THREAD

**Issue by tootbrute** [simplex-chat/simplex-chat#1996]
I am using SimpleX on three phones but sometimes messages don't get through to both sides:

A. Asus Zenphone Max M2 (ASUS_X01BDA), PixelExperience-Android13, SimpleX v4.5.3 (102)
B. Samsung Galaxy A7050, LineageOS 20, no Google, SimpleX v4.5.3 (102)
C. Samsung A53 (stock Google)

A and B work together great.
C to A is no problem, too.
C to B doesn't always work. Sometimes C sends messages and I don't recieve them. They get all my messages.

I am using my own server as the 'send' server.
All these phones are on the same Tailscale network.

- How to toubleshoot these sorts of issues?
- What settings might help make message sending more reliable?
- Any specific server settings that might help?

Currently, my XMPP server is more reliable than my SimpleX server and my significant other is not happy when she sends me messages from C and I don't see them. Right now, messages are generally more reliable than Sessions right now, but not as good as Signal and that's the gold standard I think you are aiming for.

**Comment by epoberezkin** [comment]
how reliable is "test" of the servers you have configured in the app?

**Comment by tootbrute** [comment]
> how reliable is "test" of the servers you have configured in the app?
Not sure what you mean.  I went in the app and did test of SMP servers. All good. I just went to do test of XFTP servers and some fail.

On my school wifi: xftp1.simplex.im fails. The xftp2-6 worked. I clicked reset and tested servers again. It worked.
On my 

NOTE: I updated my issue with current information at top too.

- I stopped using my own server.
- C is not on Tailscale anymore.

I've had good success with the apps for the past  month or so but this week I have issues again. B sending to C not seeing messages. C sending to B not seeing messages.

Are there options to set to make things more reliable, even if it cuts down the speed? 
I already made TCP connection timeout 20 seconds. I'm thinking I should increase protocol timeout?

**Comment by epoberezkin** [comment]
yes, protocol timeout is what is commonly triggered - TCP connection is only used on the initial connection.
